### PR TITLE
Fix for new user creation

### PR DESCRIPTION
**Pull request by Vladislav-Kisliy**

### CODE REVIEW

File: backend/src/main/java/com/messenger/backend/controllers/UserController.java [hunk 1]
```diff
@@ -40,10 +40,16 @@ public Boolean getUserStatus(@RequestParam(value = "id") Integer id) {
         return userService.getUserStatus(id);
     }
 
-    @PutMapping(value = "/updateUser", consumes = MediaType.APPLICATION_JSON_VALUE) //update contact list
-
+    @GetMapping(value = "/createUser", consumes = MediaType.APPLICATION_JSON_VALUE)
```
Comment (vosmerkin) [inline backend/src/main/java/com/messenger/backend/controllers/UserController.java:43]: In CRUD operations, "Create" is implemented by POST method. But, as I understand, using more simple method is acceptable. Especcially in our case, where the only parameter passed is newUserName (String)

File: backend/src/main/java/com/messenger/backend/services/UserService.java [hunk 2]
```diff
@@ -13,25 +16,30 @@ public class UserService {
...
 
+    public UserEntity createUser(String userName) {
+        UserEntity newUser = new UserEntity(null, userName, new Date(), true, Collections.emptySet());
+        return userRepository.save(newUser);
+    }
 
     public UserEntity updateContactList(UserEntity userEntity) {
         Integer id = userEntity.getId();
+        UserEntity result = UserEntity.EMPTY_ENTITY;
         if (userRepository.existsById(id)) {
```
Comment (vosmerkin) [inline backend/src/main/java/com/messenger/backend/services/UserService.java:39]: Should we test with "existsById(id)" everywhere? For example in getUserStatus method?

File: backend/src/main/resources/application-local.properties [hunk 3]
```diff
@@ -1,3 +1,3 @@
 spring.datasource.url=jdbc:postgresql://localhost:5432/messenger
 spring.datasource.username=postgres
-spring.datasource.password=Ev222322
+spring.datasource.password=UPS
```
Comment (vosmerkin) [inline backend/src/main/resources/application-local.properties]: I guess this file is also shoud be removed

File: ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java [hunk 4]
```diff
@@ -17,43 +17,48 @@
...
+    private final String backendHost;
+    private final String userCreateAddress;
+    private final String roomCreateAddress;
+    private final String userInfo;
 
     public HttpBackendClient() {
+        client = HttpClient.newHttpClient();
+        backendHost = PropertyManager.getProperty("backend.host");
+        userCreateAddress = backendHost + PropertyManager.getProperty("backend.user_create");
+        roomCreateAddress = PropertyManager.getProperty("backend.room_create");
```
Comment (vosmerkin) [inline ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java]: Maybe these variables should be  of the same format? (= backendHost +   xxx)

File: ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java [hunk 5]
```diff
@@ -17,43 +17,48 @@
...
         try {
             response = client.send(request, HttpResponse.BodyHandlers.ofString());
         } catch (IOException e) {
             throw new HttpClientIOException("IOException to remote address " + userRequestAddress);
         }
-        resultString = response.body();
-        if (response.statusCode() == 404) throw new UserNotFoundException("User '" + userName + "' not found");
-        result = JsonMapper.fromJson(resultString, UserDto.class);
-        return result;
+        if (response.statusCode() > 404) throw new UserNotFoundException("User '" + userName + "' not found");
```
Comment (vosmerkin) [inline ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java]: Status code 404 - Not_Found

File: ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java [hunk 6]
```diff
@@ -67,31 +72,31 @@ public UserDto userUpdate(UserDto userDto) throws IOException, InterruptedExcept
...
 
-    public UserDto userCreate(NewUserDto newUserDto) throws InterruptedException, IOException {
-        UserDto userDto;
-        String resultString;
-        String userCreateAddress = PropertyManager.getProperty("backend.user_create");
-        log.debug("Connecting remote host {}", backendHost + userCreateAddress);
-        request = HttpRequest.newBuilder(URI.create(backendHost + userCreateAddress))
-                .header("Content-Type", "application/json")
-                .POST(HttpRequest.BodyPublishers.ofString(JsonMapper.toJson(newUserDto)))
+    public UserDto userCreate(String userName) throws InterruptedException, IOException {
```
Comment (vosmerkin) [inline ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java:79]: I'll rebuild roomCreate method  to roomCreate(String newRoom)

File: ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java [hunk 7]
```diff
@@ -67,31 +72,31 @@ public UserDto userUpdate(UserDto userDto) throws IOException, InterruptedExcept
...
             response = client.send(request, HttpResponse.BodyHandlers.ofString());
         } catch (IOException e) {
+            log.error("User creation has failed: {}", e.getMessage());
             throw new HttpClientIOException("IOException to remote address " + userCreateAddress);
         }
-        resultString = response.body();
-        userDto = JsonMapper.fromJson(resultString, UserDto.class);
-        return userDto;
+        if (response.statusCode() > 399) {
+            throw new HttpClientIOException("IOException to remote address " + userCreateAddress);
```
Comment (vosmerkin) [inline ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java]: If we have response - there is no problem with connection. Its some kind of backend error. Ill make BackendErrorException

File: ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java (hunk 4 above)
Comment (Vladislav-Kisliy) [reply ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java]: I didn't get you, it's only refactoring existing code. Do you see roomCreateAddress as backendHost + xxx ?

Comment (vosmerkin) [reply ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java]: I propose to do like this:
`       userCreateAddress = backendHost + PropertyManager.getProperty("backend.user_create");
        roomCreateAddress = backendHost + PropertyManager.getProperty("backend.room_create");`

Comment (Vladislav-Kisliy) [reply ui/src/main/java/com/messenger/ui/services/HttpBackendClient.java]: ok, sure

File: backend/src/main/java/com/messenger/backend/entity/UserEntity.java [hunk 8]
```diff
@@ -55,11 +66,37 @@ public Integer getId() {
...
     public Set<UserEntity> getContactList() {
         return contactList;
     }
 
     public void setContactList(Set<UserEntity> contactList) {
         this.contactList = contactList;
     }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (vosmerkin) [inline backend/src/main/java/com/messenger/backend/entity/UserEntity.java:79]: Methods generated by IDEA

File: backend/src/main/java/com/messenger/backend/services/UserService.java [hunk 9]
```diff
@@ -13,25 +16,30 @@ public class UserService {
...
+
     public UserEntity getByUserName(String userName) {
         return userRepository.findByUserName(userName);
     }
 
     public Boolean getUserStatus(Integer id) {
-        Boolean status =false;
         UserEntity user = userRepository.findAllById(id);
-        status = user.getActiveStatus();
+        boolean status = user.getActiveStatus();
```
Comment (vosmerkin) [inline backend/src/main/java/com/messenger/backend/services/UserService.java:27]: Is it too bad to split declaration and assignment?

File: backend/src/main/java/com/messenger/backend/services/UserService.java [hunk 10]
```diff
@@ -13,25 +16,30 @@ public class UserService {
...
     }
 
+    public UserEntity createUser(String userName) {
+        UserEntity newUser = new UserEntity(null, userName, new Date(), true, Collections.emptySet());
+        return userRepository.save(newUser);
+    }
 
     public UserEntity updateContactList(UserEntity userEntity) {
         Integer id = userEntity.getId();
+        UserEntity result = UserEntity.EMPTY_ENTITY;
```
Comment (vosmerkin) [inline backend/src/main/java/com/messenger/backend/services/UserService.java:38]: so when userRepository.existsById(id)=null we return EMPTY_ENTITY and make also return status code 404? Or just one of these?

**Pull request merged**